### PR TITLE
1.3

### PR DESCRIPTION
**Pull request by maZahaca**
- Major bug fixes
- Tested capability with symfony 2.7.*
- Added checking of controller existence inside of ActionResolver to improve the code
- CodeStyle moved to PSR2

### CODE REVIEW

**Comment by dariuszz123** [comment]
Thanks for PR!

File: CiActionResolveEvent.php [hunk 1]
```diff
@@ -19,54 +19,60 @@
...
      * @param string $controller
-     * @param string $method 
+     * @param string $method
+     * @param string $locale
      */
-    public function addPossibleAction($controller, $method, $locale) {
-        $this->possible_methods[] = array(
+    public function addPossibleAction($controller, $method, $locale)
+    {
+        $this->possibleMethods[] = [
```
Comment (tadcka) [inline CiActionResolveEvent.php]: php 5.3 support...

File: CiHelperService.php [hunk 2]
```diff
@@ -136,58 +150,68 @@ public function setCiPaths(Request $request = null) {
...
-                }
-
-                define('APPPATH', BASEPATH . $application_folder . '/');
-            }
-
-            $this->paths_initalized = true;
+            // Path to the front controller (this file)
+            define('FCPATH', $applicationFolder.'../');
+
+//            if (!defined('APPPATH')) {
```
Comment (tadcka) [inline CiHelperService.php]: remove comment code?

**Comment by maZahaca** [comment]
I'll fix it in the next PR, added notes to myself.
Thanks for feedback.

File: DefaultCiActionResolver.php [hunk 3]
```diff
@@ -59,7 +71,15 @@ public function addPossibleRoutes(CiActionResolveEvent $event, &$pathParts, $ind
     public function onActionResolveEvent(CiActionResolveEvent $event)
     {
         $path = $event->getRequest()->getPathInfo();
-        $parts = explode('/', substr($path, 1));
+        $part = (string) substr($path, 1);
+        if ($part === '') {
+            $part = 'home/';
```
Comment (Nercury) [inline DefaultCiActionResolver.php]: Why support this `/home`? I think it should be left to application to redirect to appropriate home page.

Comment (maZahaca) [inline DefaultCiActionResolver.php]: As I remember I got a problem with index page, my code igniter didn't return it as a route and symfony will show a 404 page.
I will investigate it again. And will answer directly

Comment (Nercury) [inline DefaultCiActionResolver.php]: Let's add this only if CodeIgniter itself does it.

Comment (maZahaca) [inline DefaultCiActionResolver.php]: Yep

**Comment by tadcka** [comment]
:+1:

**Pull request merged**